pke/kyber/internal/common/ntt_test.go

```diff
@@ -34,14 +35,20 @@ func BenchmarkInvNTTGeneric(b *testing.B) {
 }
 
 func (p *Poly) Rand() {
+	pp := make([]uint8, 4*N)
+	_, _ = rand.Read(pp)
+	max := uint32(Q)
 	for i := 0; i < N; i++ {
-		p[i] = int16(mathRand.Intn(int(Q)))
+		p[i] = int16(binary.LittleEndian.Uint32(pp[4*i:]) % max)
 	}
 }
 
 func (p *Poly) RandAbsLeQ() {
+	pp := make([]uint8, 4*N)
+	_, _ = rand.Read(pp)
+	max := 2 * uint32(Q)
 	for i := 0; i < N; i++ {
-		p[i] = int16(mathRand.Intn(int(2*Q))) - Q
+		p[i] = int16(int32(binary.LittleEndian.Uint32(pp[4*i:])%max) - int32(Q))
```

Comment:
This loop gets plopped down a lot. Is it possible to ractor the code so that the common functionality is implemented by a single function? 
```go
func setRandPoly(p []int16) err {
        pp := make([]uint8, 4*N)
	if n, err := rand.Read(pp); err != nil {
            panic!(err)
        else if n < len(pp) {
            panic!("short read from RNG")
	}
	max := 2 * uint32(Q)
	for i := 0; i < N; i++ {
		p[i] = int16(int32(binary.LittleEndian.Uint32(pp[4*i:])%max)
        }
}
```

Reply:
nice suggestion, refactored now